bundle/edu.gemini.oodb.too.url/src/main/java/edu/gemini/spdb/rapidtoo/impl/TooUpdateServiceImpl.java

```diff
@@ -236,9 +237,12 @@ private void _setTarget(ISPObservation obs, TooUpdate update)  {
         TargetEnvironment targetEnv = targetObsComp.getTargetEnvironment();
         SPTarget base = targetEnv.getBase();
 
-        base.setName(tooTarget.getName());
-        base.setRaDecDegrees(tooTarget.getRa(), tooTarget.getDec());
-        base.setMagnitudes(tooTarget.getMagnitudes());
+        base.getTarget().setName(tooTarget.getName());
+        base.notifyOfGenericUpdate();
+        base.getTarget().getRa().setAs(tooTarget.getRa(), CoordinateParam.Units.DEGREES);
+        base.getTarget().getDec().setAs(tooTarget.getDec(), CoordinateParam.Units.DEGREES);
+        base.notifyOfGenericUpdate();
+        base.getTarget().setMagnitudes(tooTarget.getMagnitudes());
```

Comment:
I suppose it is irrelevant given where this is going, but why not just a single `base.notifyOfGenericUpdate()` at the end?